lib/ButtonNew/ButtonBase.js

```diff
@@ -29,6 +29,10 @@ function ButtonBase({
     }
   );
 
+  if (typeof children === 'function') {
```

Comment:
💚 I worry a little that we may be under considering what adding this means.

If we pass the a function as a child, the loader props all become useless, which feels like bad design.

I definitely think that we should be able to use different element types for buttons, but this reminds me of our old button component, which was built to do just about everything in a single component, and as a result ended up being a pain in the ass to maintain.

I'm not 100% on the answer here, so this is not a blocker!

Reply:
💚 Should we add a story to show the button styles being applied to an anchor?